No Makefile! Upstream says to execute
cc -o inform *.c
So let's make a simple Makefile

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -0,0 +1,12 @@
+# Inform6 compiler Makefile
+# Written by Brian Callahan <bcallah@openbsd.org>
+#   and released into the Public Domain
+
+PROG =	inform
+OBJS =	arrays.o asm.o bpatch.o chars.o directs.o errors.o \
+	expressc.o expressp.o files.o inform.o lexer.o \
+	linker.o memory.o objects.o states.o symbols.o \
+	syntax.o tables.o text.o veneer.o verbs.o
+
+all: ${OBJS}
+	${CC} ${LDFLAGS} -o ${PROG} ${OBJS}
